inductiva/tasks/task.py

```diff
@@ -172,13 +172,64 @@ def wait(self, polling_period: int = 5) -> models.TaskStatusCode:
 
             time.sleep(polling_period)
 
-    def kill(self) -> None:
-        """Kill the task.
-
-        This method issues a request to the API to kill the task. It doesn't
-        block waiting for confirmation if the task was killed.
+    def kill(self,
+             wait: bool = False,
+             timeout: float = 5,
+             maxretries: int = 5) -> bool:
+        """Tries to kill a task.
+        If wait is False (default value) the function sends the
+        kill command and returns. If wait is True the function
+        checks for confirmation that the kill command was successfull.
+        Args:
+            wait: Determines if we wait for the kill command to
+                finish.
+            timeout: Time in seconds to wait for the kill command.
+            maxretries: Number of times to retry the kill command.
+        Returns:
+            True if the kill command was successfull, False otherwise.
         """
-        self._api.kill_task(path_params=self._get_path_params())
+        while maxretries > 0:
+            maxretries -= 1
+            try:
+                self._api.kill_task(path_params=self._get_path_params())
```

Comment:
Moreover, either we check that maxretries should be > -1 to handle one kill task at least, but still check that  `maxretries>=0`.

Reply:
I now raise a ValueError exception.